library/std/src/sys/windows/process.rs

```diff
@@ -700,13 +699,15 @@ enum Quote {
 
 // Produces a wide string *without terminating null*; returns an error if
 // `prog` or any of the `args` contain a nul.
-fn make_command_line(prog: &OsStr, args: &[Arg], force_quotes: bool) -> io::Result<Vec<u16>> {
+fn make_command_line(prog: &[u16], args: &[Arg], force_quotes: bool) -> io::Result<Vec<u16>> {
     // Encode the command and arguments in a command line string such
     // that the spawned process may recover them using CommandLineToArgvW.
     let mut cmd: Vec<u16> = Vec::new();
-    // Always quote the program name so CreateProcess doesn't interpret args as
-    // part of the name if the binary wasn't found first time.
-    append_arg(&mut cmd, prog, Quote::Always)?;
+    // Always quote the program name so CreateProcess to avoid ambiguity when
+    // the child process parses its arguments.
+    cmd.push(b'"' as u16);
+    cmd.extend_from_slice(prog.strip_suffix(&[0]).unwrap_or(prog));
```

Comment:
This doesn't escape `\` and `"` anymore.

Reply:
That is intentional. The [zeroth argument is special](https://docs.microsoft.com/en-us/cpp/cpp/main-function-command-line-args?view=msvc-160#parsing-c-command-line-arguments):

> The first argument (argv[0]) is treated specially. It represents the program name. Because it must be a valid pathname, parts surrounded by double quote marks (") are allowed. The double quote marks aren't included in the argv[0] output. The parts surrounded by double quote marks prevent interpretation of a space or tab character as the end of the argument. The later rules in this list don't apply.

Essentially argv[0] does not have escape characters but that's ok because [`"` is not allowed in Windows paths](https://docs.microsoft.com/en-us/windows/win32/fileio/naming-a-file#naming-conventions). See also, our parsing of the command line:

https://github.com/rust-lang/rust/blob/4fb45245840ad129b305023995a886cf8ec3e279/library/std/src/sys/windows/args.rs#L73-L86)